pages/level-change.js

```diff
@@ -0,0 +1,274 @@
+import Link from "next/link";
+import Layout from "../components/layout.js";
+import styled, { css } from "styled-components";
+
+import { getUsers } from "../networkRequests/user";
+import { roleChange } from "../networkRequests/roles";
+class LevelChangePage extends Component {
+  state = {
+    rolesDropDownOpen: -1,
+    loading: false,
+    roles: ["grand wizard", "super asshole", "big dick energy"],
+    users: [],
+    submitting: false,
+    success: false,
+    error: null
+  };
+
+  async componentWillMount() {
+    this.setState({ loading: true });
+    const usersResponse = await getUsers();
+    console.log(usersResponse);
+    if (usersResponse.success) {
+      this.setState({
+        users: usersResponse.users,
+        loading: false,
+        roles: usersResponse.roles
+      });
+    } else if (usersResponse.error && !!usersResponse.error.message) {
+      this.setState({ eror: usersResponse.error.message, loading: false });
+    } else {
+      this.setState({
+        error: "there was an error with the request",
+        loading: false
+      });
+    }
+  }
+
+  handleClick = (idx, e) => {
+    if (idx === this.state.rolesDropDownOpen) {
+      this.setState({ rolesDropDownOpen: -1 });
+    } else {
+      this.setState({ rolesDropDownOpen: idx });
+    }
+  };
+  clearSuccessError = () => {
+    this.setState({ error: "", success: false });
+  };
+
+  changeRole = async (userID, role, e) => {
+    this.setState({ submitting: true });
+    console.log("boi");
+    const roleChangeResponse = await roleChange(userID, role);
+    console.log("huh");
+    console.log(roleChangeResponse);
+    if (roleChangeResponse.success) {
+      this.setState({ submitting: false, success: true });
+    } else if (!!roleChangeResponse.error) {
+      this.setState({
+        submitting: false,
+        error: roleChangeResponse.error,
+        loading: false
+      });
+    } else {
+      this.setState({
+        submitting: false,
+        error: "there was an error with the request",
+        loading: false
+      });
+    }
+  };
+
+  render() {
+    const { loading, users, submitting, success, error, roles } = this.state;
+
+    const showModal = loading || submitting || success || !!error;
+    console.log(showModal, error);
+    const modalText = loading
+      ? "Loading..."
+      : submitting
+      ? "Submitting..."
+      : success
+      ? "Success!"
+      : error;
+    return (
+      <React.Fragment>
+        <Page showModal={showModal}>
+          <Header>
+            <Text size="2rem">Level Change Users!</Text>
+          </Header>
+          {!!users && !!users.length ? (
+            <React.Fragment>
+              <Text style={{ margin: "1rem" }} size="2rem">
+                Users
+              </Text>
+              <UsersList>
+                {users.map((user, idx) => {
+                  return (
+                    <Row
+                      children={
+                        <RolesDropDown
```

Comment:
I would suggest looking into https://reactstrap.github.io/components/dropdowns/ and using their dropdowns instead of trying to style and use our own. It just prevents us from reinventing the wheel in terms of styling and creating dropdowns menus.